### PR TITLE
Use Composer 2

### PR DESCRIPTION
**Pull request by Taluu**
Or maybe we could parametrize it in the config file ?

### CODE REVIEW

**Comment by Taluu** [comment]
Not sure if what I tried works. We could also make the composer version an argument instead ?

File: Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ LABEL "maintainer"="Matt Brown <github@muglug.com>"
 
 RUN apk add --no-cache tini git
 
-COPY --from=composer:1.9 /usr/bin/composer /usr/bin/composer
+COPY --from=composer:$INPUT_COMPOSER_VERSION /usr/bin/composer /usr/bin/composer
```
Comment (Taluu) [inline Dockerfile:18]: not sure if this is how we can access the input from github action ?

Comment (Taluu) [reply Dockerfile]: Nevermind, as the image will be built once I think and not on the fly according to the args of the action.

**Comment by jderusse** [comment]
Any blocker for this PR?

File: Dockerfile [hunk 2]
```diff
@@ -15,18 +15,19 @@ LABEL "maintainer"="Matt Brown <github@muglug.com>"
 
 RUN apk add --no-cache tini git
 
-COPY --from=composer:1.9 /usr/bin/composer /usr/bin/composer
+COPY --from=composer:2 /usr/bin/composer /usr/bin/composer2
+COPY --from=composer:1 /usr/bin/composer /usr/bin/composer1
```
Comment (jderusse) [inline Dockerfile:19]: Is it still needed in 2021 to expose the old composer version? I'd rather expose only the versions 2.0 and simplify/lighten this Image.

Comment (Taluu) [reply Dockerfile:19]: I wasn't sure either, but I thought that maybe people would like to keep composer1 for some stuff (e.g not having a composer plugin compatible with composer 2).

Unless we could ask them to use an old version of this action ?

Comment (jderusse) [reply Dockerfile:19]: My 2 cents: People can use an older version of this Image, or use a previous step in github workflow that run composer install from the composer version they want.

Let's spread the adoption of composer v2.

Comment (Taluu) [reply Dockerfile]: Removed that part then, forcing to use composer2 at all time.

**Review by jderusse — APPROVED** [review]

**Pull request merged**